src/gui/sdlmain.cpp

```diff
@@ -1924,23 +1936,61 @@ static bool DetectResizableWindow()
 #endif // C_OPENGL
 }
 
+static bool detect_shader_fixed_size()
+{
+#if C_OPENGL
+	if (sdl.desktop.want_type != SCREEN_OPENGL)
+		return false;
+
+	const std::string sname = get_glshader_value();
+	return (sname == "crt-fakelottes-flat" || sname == "crt-easymode-flat");
+#else
+	return false;
+#endif // C_OPENGL
+}
+
+static std::tuple<int, int> detect_window_size()
+{
+	SDL_Rect bounds;
+	SDL_GetDisplayBounds(sdl.display_number, &bounds);
+	constexpr int resolutions[][2] = {
+	        {1600, 1200},
+	        {1280, 960},
+	        {1024, 768},
+	        {800, 600},
```

Comment:
Are these DOS-side resolutions?

Reply:
No, these are user screen size resolutions - e.g. if user's desktop area is big enough to use window size 1600x1200, then we pick this as the default window size (only when `windowresolution = default` and specific shaders are selected).

Reply:
Thanks for the explanation (could that be added as a comment?)
These handle the the 4k range:
```
{2560, 1920}
{2400, 1800}
{2048, 1536}
{1920, 1440}
```

Reply:
Hmm, I'll add these resolutions but commented-out for now. And I commented-out 1600x1200 for now as well.

We want this to work, but I have no idea how this will behave for users who have pixel-doubling / display scaling enabled on OS level. This will require *extensive* testing on Windows, Linux, and macOS :( 

Resolutions 1280x960 and below are fine - even if OS will double the window size, users of 4K screens will end up with 2560x1920 window which will still fit on screen.

We can't depend on it working correctly now, because SDL2 had some issues with high density displays IIRC and we have non-SDL2 hack for pixel doubling on Windows inherited from our upstream.

Reply:
Good catch with the line/pixel doubling. I think I've seen this in some modes, and it blows out the image so you only see a section of the DOS display. This is a nice safe subset for now.